firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionImpl.java

```diff
@@ -297,6 +306,43 @@ private UpdateTask updateApp(boolean showDownloadInNotificationManager) {
     }
   }
 
+  @Override
+  public void collectAndSendFeedback() {
+    collectAndSendFeedback(Executors.newSingleThreadExecutor());
+  }
+
+  @VisibleForTesting
+  public void collectAndSendFeedback(Executor taskExecutor) {
+    screenshotTaker
+        .takeScreenshot()
+        .onSuccessTask(
+            taskExecutor,
+            screenshot ->
+                testerSignInManager
+                    .signInTester()
+                    .addOnFailureListener(
+                        taskExecutor,
+                        e ->
+                            LogWrapper.getInstance()
+                                .e("Failed to sign in tester. Could not collect feedback.", e))
+                    .onSuccessTask(taskExecutor, unused -> releaseIdentifier.identifyRelease())
```

Comment:
I don't think so. I assume we want to take the screenshot ASAP, as opposed to potentially having to wait until they visit the browser to sign in. And then we have to sign them in before trying to identify the release, since that requires an API call. We _could_ look up the binary identifier in parallel to the screenshot or sign in, but I'm not sure it's worth it to complicate the ReleaseIdentifier API by splitting the binary identifier computation and API call.